spec/60-trace-id-format.md

```diff
@@ -1,20 +1,67 @@
-a value like `4bf92f3577b34da6a3ce929d0e0e4736` where `4bf92f3577b34da6a` is a
-random value or a function of time and host value.
-
-**Note**: Even though a system may operate with a shorter `trace-id` for
-[distributed trace](https://w3c.github.io/trace-context/#dfn-distributed-traces)
-reporting, the full `trace-id` MUST be propagated to conform to the
-specification.
+This section suggests some best practices to consider when platform or tracing
+vendor implement `trace-id` generation and propagation algorithms. These
+practices will ensure better interoperability of different systems.
+
+### Uniqueness of `trace-id`
+
+The value of `trace-id` SHOULD be globally unique. This field is typically used
+for unique identification of a <a>distributed trace</a>. It is common for
+<a>distributed traces</a> to span various components, including, for example,
+cloud services. Cloud services tend to serve variety of clients and have a very
+high throughput of requests. So global uniqueness of `trace-id` is important,
+even when local uniqueness might seem like a good solution.
+
+### Randomness of `trace-id`
+
+Randomly generated value of `trace-id` SHOULD be preferred over other
+algorithms of generating a globally unique identifiers. Randomness of `trace-id`
+addresses some [security](#security-considerations) and [privacy
+concerns](#privacy-considerations) of exposing unwanted information. Randomness
+also allows tracing vendors to base sampling decisions on `trace-id` field value
+and avoid propagating an additional sampling context.
+
+As shown in the next section, it is important for `trace-id` to carry
+"uniqueness" and "randomness" in the right part of the `trace-id`, for better
+inter-operability with some existing systems.
+
+### Handling `trace-id` for complaint platforms with shorter internal identifiers
+
+There are tracing systems which use a `trace-id` that is shorter than 16 bytes,
+which are still willing to adopt this specification.
+
+If such a system is capable of propagating a fully compliant `trace-id`, even
+while still requiring a shorter, non-complaint identifier for internal purposes,
+the system is encouraged to utilize the `tracestate` header to propagate the
+additional internal identifier. However, if a system would instead prefer to use
+the internal identifier as the basis for a fully complaint `trace-id`, it SHOULD
+be incorporated at the as rightmost part of a `trace-id`. For example, tracing
+system may receive `234a5bcd543ef3fa53ce929d0e0e4736` as a `trace-id`, hovewer
+internally it will use `53ce929d0e0e4736` as an identifier.
+
+### Interoperating with existing systems which use shorter identifiers
+
+There are tracing systems which are not capable of propagating the entire 16
+bytes of a `trace-id`. For better interoperability between a fully compliant
+systems with these existing systems, the following practices are recommended:
+
+1. Whenever a shorter `trace-id` needs to be send, a compliant `trace-id` SHOULD
```

Comment:
sorry, I disagree with @tedsuo's comments on ingress/egress. The responsibility for conversion should be on the non-compliant system, not on the compliant sender/receiver. If that system is a blackbox that cannot be changed, it can be deployed with proxies / sidecar that will perform rewriting of the headers to the format it understands. But if my service is fully compliant and needs to talk to such legacy system, I should not be required to couple my service with understanding of what the legacy system understands or not.

Reply:
This is why I was suggesting to phrase these in terms of ingress/egress to/from non-compliant region of the architecture.

Reply:
the word compliant here refers to compliant trace-id. I can replace it with 16 bytes `trace-id`

Reply:
I think @tedsuo wrote the paragraph with the idea on how to convert from traceparent to some other format with shorter ID. This is why it's harder to read with the different idea in mind. I added paragraph to talk about conversions

Reply:
added clarifications and rephrased a bit